### PR TITLE
STAF-21/27/29: add login and notification services with RTK Query

### PR DESCRIPTION
**Pull request by devmc-ee**

### CODE REVIEW

File: src/core/types/notifications.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export enum NotificationLevelType {
+  SUCCESS = 'success',
+  ERROR = 'error',
+}
+
+export interface NotificationObject {
+  id: string;
+  message: string;
+  level: NotificationLevelType;
+}
```
[the remark this answers is not in the document]
Comment (devmc-ee) [reply src/core/types/notifications.d.ts:10]: Good catch, thanks! Fixed!

File: src/core/components/NavMenu.tsx [hunk 2]
```diff
@@ -16,6 +19,8 @@ export const NavMenu = (): JSX.Element => {
     return isActiveRoute ? activeRouteClassName : '';
   };
 
+  const logout = () => dispatch(clearUserDate());
```
Comment (RomanLavrunov) [inline src/core/components/NavMenu.tsx:22]: Just to make sure it's 100% clear that we erase user data or date? USER_DATA_CLEARED: clearUserDate.

File: src/user/redux/userActions.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const USER_ACTION_TYPE = {
+  LOGIN: 'user/LOGGED_IN',
+};
+
+export const userAction = {
+  login: (token: string) => ({
+    type: USER_ACTION_TYPE.LOGIN,
+    payload: { token },
+  }),
+};
```
Comment (RomanLavrunov) [inline src/user/redux/userActions.ts:10]: This piece of code seemed to have been left unused. There would probably be zero use of keeping it at the project at the moment.

File: src/core/components/NavMenu.tsx (hunk 2 above)
Comment (devmc-ee) [reply src/core/components/NavMenu.tsx]: Typo, fixed!

File: src/user/redux/userActions.ts (hunk 3 above)
Comment (devmc-ee) [reply src/user/redux/userActions.ts]: Thanks! Removed

**Review by RomanLavrunov — APPROVED** [review]

**Pull request merged**